----We are looking at Titanic Data Set----
----Objective:
----Determine the main survival indicators
----Predict the same with a model

In [2]:
import pandas as pd

In [3]:
#Read CSV file
df = pd.read_csv(r'C:\Users\Abhijith N S\Desktop\ML Projects\ABIBI\1. Titanic Problem Statement\Datasets\tested.csv')

In [4]:
df.head(5)

PassengerId  Survived  Pclass  \
0          892         0       3   
1          893         1       3   
2          894         0       2   
3          895         0       3   
4          896         1       3   

                                           Name     Sex   Age  SibSp  Parch  \
0                              Kelly, Mr. James    male  34.5      0      0   
1              Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                     Myles, Mr. Thomas Francis    male  62.0      0      0   
3                              Wirz, Mr. Albert    male  27.0      0      0   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   

    Ticket     Fare Cabin Embarked  
0   330911   7.8292   NaN        Q  
1   363272   7.0000   NaN        S  
2   240276   9.6875   NaN        Q  
3   315154   8.6625   NaN        S  
4  3101298  12.2875   NaN        S

--- Information on Columns
survival - Survival (0 = No; 1 = Yes)
class - Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd)
name - Name
sex - Sex
age - Age
sibsp - Number of Siblings/Spouses Aboard
parch - Number of Parents/Children Aboard
ticket - Ticket Number
fare - Passenger Fare
cabin - Cabin
embarked - Port of Embarkation (C = Cherbourg; Q = Queenstown; S = Southampton)

In [5]:
df.shape

(418, 12)

There are 418 obs we have. 
We have 152 indivisual survided and 266 died 

In [6]:
df['Survived'].value_counts()

0    266
1    152
Name: Survived, dtype: int64

In [8]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   418.000000  418.000000  418.000000  332.000000  418.000000   
mean   1100.500000    0.363636    2.265550   30.272590    0.447368   
std     120.810458    0.481622    0.841838   14.181209    0.896760   
min     892.000000    0.000000    1.000000    0.170000    0.000000   
25%     996.250000    0.000000    1.000000   21.000000    0.000000   
50%    1100.500000    0.000000    3.000000   27.000000    0.000000   
75%    1204.750000    1.000000    3.000000   39.000000    1.000000   
max    1309.000000    1.000000    3.000000   76.000000    8.000000   

            Parch        Fare  
count  418.000000  417.000000  
mean     0.392344   35.627188  
std      0.981429   55.907576  
min      0.000000    0.000000  
25%      0.000000    7.895800  
50%      0.000000   14.454200  
75%      0.000000   31.500000  
max      9.000000  512.329200

In [10]:
df[df.Survived == 1].describe()

PassengerId  Survived      Pclass         Age       SibSp       Parch  \
count   152.000000     152.0  152.000000  127.000000  152.000000  152.000000   
mean   1096.789474       1.0    2.144737   30.272362    0.565789    0.598684   
std     123.345508       0.0    0.887051   15.428613    0.974313    1.105434   
min     893.000000       1.0    1.000000    0.170000    0.000000    0.000000   
25%     989.500000       1.0    1.000000   20.500000    0.000000    0.000000   
50%    1093.500000       1.0    2.000000   27.000000    0.000000    0.000000   
75%    1205.250000       1.0    3.000000   38.500000    1.000000    1.000000   
max    1306.000000       1.0    3.000000   76.000000    8.000000    9.000000   

             Fare  
count  152.000000  
mean    49.747699  
std     73.108716  
min      6.950000  
25%      8.626050  
50%     21.512500  
75%     55.441700  
max    512.329200

In [11]:
df[df.Survived == 0].describe()

PassengerId  Survived      Pclass         Age       SibSp       Parch  \
count   266.000000     266.0  266.000000  205.000000  266.000000  266.000000   
mean   1102.620301       0.0    2.334586   30.272732    0.379699    0.274436   
std     119.520499       0.0    0.808497   13.389528    0.843735    0.883745   
min     892.000000       0.0    1.000000    0.330000    0.000000    0.000000   
25%     999.250000       0.0    2.000000   22.000000    0.000000    0.000000   
50%    1103.500000       0.0    3.000000   27.000000    0.000000    0.000000   
75%    1203.750000       0.0    3.000000   40.000000    1.000000    0.000000   
max    1309.000000       0.0    3.000000   67.000000    8.000000    9.000000   

             Fare  
count  265.000000  
mean    27.527877  
std     41.079423  
min      0.000000  
25%      7.854200  
50%     13.000000  
75%     26.550000  
max    262.375000